omx/gstomxaacdec.c

```diff
@@ -96,70 +94,114 @@ gst_omx_aac_dec_set_format (GstOMXAudioDec * dec, GstOMXPort * port,
   const gchar *stream_format;
 
   gst_omx_port_get_port_definition (port, &port_def);
-  port_def.format.audio.eEncoding = OMX_AUDIO_CodingAAC;
+
+  if (port->index == GST_OMX_AAC_DEC_INPUT_PORT) {
+    port_def.nPortIndex = port->index;
```

Comment:
There are many replicated port definitions. I think its better if you factor them out and inside the `ifs` you only set the ones that are different.

Reply:
Yes, nice observation. Ready: ac664c134ac783d37aec797017d18a2765849fe3